src/json-patch-duplex.ts

```diff
@@ -524,6 +525,74 @@ module jsonpatch {
     return true;
   }
 
+  
+  /**
+     * Apply a single json-patch on an object tree
+     * Returns the result object.
+     */
+  export function applyPatch<T>(tree: T, patch: Patch<any>): T;
+  export function applyPatch<T>(tree: T, patch: Patch<any>, validate: boolean): T;
+  export function applyPatch<T>(tree: T, patch: Patch<any>, validate = false): T {
+    if (typeof validate !== 'boolean') {
+      validate = false;
+    }
+    // on root
+    if (patch.path === "") {
+      if (patch.op === 'add' || patch.op === 'replace') {// same for add or replace 
+        // a primitive value, just return it
+        if (typeof patch.value !== 'object' || patch.value === null) {
+          return patch.value;
+        } else {
+          // conflicting types [] vs {} just return the new value
+          if (_isArray(tree) !== _isArray(patch.value)) {
+            return patch.value;
+            //same type, use original apply
```

Comment:
I agree I was building a demo and not modifying the original object seems like a semantic must. Let alone the perf boost.